src/txmempool.cpp

```diff
@@ -1197,32 +1197,14 @@ bool CTxMemPool::checkAddressNonceAndFee(const CTxMemPoolEntry &pendingEntry, co
         return true;
     }
 
-    CTxMemPool::setEntries itersToRemove;
-    std::set<CTransactionRef> txsToRemove;
-    auto& txidIndex = mapTx.get<txid_tag>();
-
     auto result{true};
+    CTxMemPool::setEntries itersToRemove;
     for (auto it = range.first; it != range.second; ++it) {
         const auto& entry = *it;
-
-        if (pendingEntry.GetEVMPromisedTipFee() > entry.GetEVMPromisedTipFee()) {
-            if (entry.GetCustomTxType() == CustomTxType::EvmTx) {
-                auto txIter = mapTx.project<0>(it);
-                itersToRemove.insert(txIter);
-            } else {
-                const auto &tx = entry.GetSharedTx();
-                for (const auto &vin : tx->vin) {
-                    auto hashEntry = txidIndex.find(vin.prevout.hash);
-                    if (hashEntry != txidIndex.end() &&
-                        hashEntry->GetCustomTxType() == CustomTxType::AutoAuthPrep) {
-                        txsToRemove.insert(hashEntry->GetSharedTx());
-                        // Add auto auth and continue, this will remove the transfer
-                        // domain TX in the removeRecursive function.
-                        continue;
-                    }
-                }
-                txsToRemove.insert(tx);
-            }
+        const auto entryType = entry.GetCustomTxType();
+        if (entryType == CustomTxType::EvmTx && entryFee >= entry.GetEVMRbfMinTipFee()) {
+            auto txIter = mapTx.project<0>(it);
+            itersToRemove.insert(txIter);
 
             // We might want to set accountsViewDirty to true here but
```

Comment:
We can remove this comment now, it is redundant since we will not be removing TransferDomain TXs. The account view will not be affected by EVM removal.